### ISSUE TITLE
Typescript?

### ISSUE THREAD

**Issue by jsdevtom** [CacheControl/json-rules-engine#76]
First of all: brilliant work! :tada:

I can see that you have gone to lots of effort with the documentation and have specified the types.

I would be willing to convert the project to TypeScript for you instead of having to write type definitions. Please let me know your opinion on this as soon as you get a chance.

Thanks
Tom

**Comment by CacheControl** [comment]
Thanks for the offer @jsdevtom.  I think Typescript is a neat language.

This project I intend to keep in pure Javascript.  Writing documentation isn't a major pain point of maintaining the project, and keeping things in Javascript offers some advantages, as its popularity makes it accessible to more contributors.

If you use Typescript would like typings made available for this project, I'd love to see someone submit `json-rules-engine` entries to the [@types organization](https://www.npmjs.com/~types).

**Comment by jsdevtom** [comment]
After converting this library to TypeScript, I now understand what you meant by 

> keeping things in Javascript offers some advantages

for example how json-rules-engine utilizes a lot of dynamic types.

In case you're interested/ want the type definitions from it somehow, I ended up converting this library to typescript and called it [truegin](https://github.com/jsdevtom/truegin). I also converted the testing framework to jest.
